README.md

```diff
@@ -1,76 +1,89 @@
+### Helm
 
-If you want to retrieve the latest client dev build for OS X or Linux, download the client:
+After you have a working Kubernetes cluster, install [helm](http://helm.sh) and run the following commands to add the Deis chart repository and install Deis to your new cluster:
 
 ```console
-$ curl -sSL http://deis.io/deis-cli/install-v2-alpha.sh | bash
+helm repo add deis https://github.com/deis/charts
+helm install deis/deis-dev
 ```
 
-If you want to hack on a new feature, build the deis/workflow image and push it to a Docker
-registry. The `$DEIS_REGISTRY` environment variable must point to a registry accessible to your
-Kubernetes cluster. You may need to configure the Docker engines on your Kubernetes nodes to allow
-`--insecure-registry 192.168.0.0/16` (or the appropriate address range).
+Note that to work off the latest stable release, change the `helm install deis/deis-dev` command to `helm install deis/deis`.
 
-When you want to test changes then commit the changes to your branch and run
+## Testing Your Code
+
+When you've built your new feature or fixed a bug, make sure you've added appropriate unit tests and run `make test` to ensure your code works properly.
+
+Also, since this component is central to the platform, it's recommended that you manually test and verify that your feature or fix works as expected. To do so, ensure the following environment variables are set:
+
+* `DEIS_REGISTRY` - A Docker registry that you have push access to and your Kubernetes cluster can pull from
+  * If this is [Docker Hub](https://hub.docker.com/), leave this variable empty
+  * Otherwise, ensure it has a trailing `/`. For example, if you're using [Quay.io](https://quay.io), use `quay.io/`
+* `IMAGE_PREFIX` - The name of the repository. This defaults to `deis`, but if you don't have access to that repository, set this to a repository that you have push access to.
+* `SHORT_NAME` (optional) - The name of the image. This defaults to `controller`
+* `VERSION` (optional) - The tag of the Docker image. This defaults to the current Git SHA (the output of `git rev-parse --short HEAD`)
+
+Then, run `make deploy` to build and push a new Docker image with your changes and replace the existing one with your new one in the Kubernetes cluster. See below for an example with appropriate environment variables.
 
 ```console
-$ make deploy
+export DEIS_REGISTRY=quay.io/
+export IMAGE_PREFIX=arschles
+make deploy
 ```
 
-This will build the required docker images and push them to the registry that was configured, then update and recreate the Replication Controller.
-Give it a bit of time for the changes to go live.
+After the `make deploy` finishes, a new pod will be launched but may not be running. You'll need to wait until the pod is listed as `Running` and the value in its `Ready` column is `1/1`. Use the following command to check the Pod's status:
+
+```console
+kubectl get pod --namespace=deis | grep deis-controller
```

Comment:
We could add the `-w`/`--watch` flag here to be more user-friendly:
`kubectl get pod --namespace=deis -w | grep deis-controller`